src/test/kotlin/org/jetbrains/research/tasktracker/tracking/TaskFileHandlerTest.kt

```diff
@@ -0,0 +1,162 @@
+                " to be 2, but found ${projectToTaskToFiles[project]?.size}"
+        }
+        assert(
+            projectToTaskToFiles[project]?.contains(task1) ?: false
+        ) { "projectToTaskToFiles[project] must contain task1" }
+        assert(
+            projectToTaskToFiles[project]?.contains(task2) ?: false
+        ) { "projectToTaskToFiles[project] must contain task1" }
+        val virtualFileTask1 = "${project.basePath}/$PLUGIN_NAME/cpp/task1.cpp".getVirtualFile().toMockTask()
+        val virtualFileTask2 = "${project.basePath}/$PLUGIN_NAME/tasks/task2.kt".getVirtualFile().toMockTask()
+        assert(
+            virtualFileTask1 == task1
+        ) { "Incorrect task $virtualFileTask1 from VirtualFile" }
+        assert(
+            virtualFileTask2 == task2
+        ) { "Incorrect task $virtualFileTask2 from VirtualFile" }
+    }
+
+    fun testInitExistingTask() {
+        TaskFileHandler.initTask(project, task1)
+        assertFailsWith<Throwable>("Expected Throwable on second initialization of $task1") {
+            TaskFileHandler.initTask(project, task1)
+        }
+    }
+
+    fun testDisposeSingleTask() {
+        TaskFileHandler.initTask(project, task1)
+        assertNoThrowable {
+            TaskFileHandler.disposeTask(project, task1)
+        }
+        assert(projectToTaskToFiles.isEmpty()) { "projectToTaskToFiles expected to be empty" }
+    }
+
+    fun testDisposeMultipleTask() {
+        TaskFileHandler.initTask(project, task1)
+        TaskFileHandler.initTask(project, task2)
+        assertNoThrowable {
+            TaskFileHandler.disposeTask(project, task1)
+        }
+        assert(
+            projectToTaskToFiles[project]?.let { it.size == 1 } ?: false
+        ) {
+            "Expected the size of 'projectToTaskToFiles' to be 1, but found ${projectToTaskToFiles.size}"
+        }
+        assertNoThrowable {
+            TaskFileHandler.disposeTask(project, task2)
+        }
+        assert(projectToTaskToFiles.isEmpty()) { "projectToTaskToFiles expected to be empty" }
+    }
+
+    fun testTaskDocumentListener() {
+        val directory = File("${project.basePath}/$PLUGIN_NAME/cpp/")
+        directory.mkdirs()
+        val virtualFile = "${directory.path}/task.cpp".getVirtualFile()
+        val document = FileDocumentManager.getInstance().getDocument(virtualFile)
+            ?: error("document with path ${virtualFile.path} must exist")
+        val logFileName =
+            "${virtualFile.nameWithoutExtension}_${virtualFile.hashCode()}_${document.hashCode()}_0.csv"
+        val logFile = File("${MainTaskTrackerConfig.pluginFolderPath}/$logFileName")
+        assert(!logFile.exists()) {
+            "log file with path ${logFile.path} should be created on first event in TaskDocumentListener"
+        }
+        document.addDocumentListener(TaskDocumentListener())
+        ApplicationManager.getApplication().runWriteAction {
+            document.setText("test")
```

Comment:
I think we need to create a helper function that adds a new text to the document (including deleting everything) and checks if this text was added. Next, call it several times for several inputs, and finally, check that all changes are in the log file. What do you think? Because now you checked only a simple case